src/room/VideoPreview.tsx

```diff
@@ -75,21 +60,76 @@ export function VideoPreview({
     settingsModalState.open();
   }, [settingsModalState]);
 
+  // Fetch user media devices.
+  const mediaDevices = useMediaDevices();
+
+  // Create local media tracks.
+  const [videoEnabled, setVideoEnabled] = React.useState<boolean>(true);
+  const [audioEnabled, setAudioEnabled] = React.useState<boolean>(true);
+  const [videoId, audioId] = [
+    mediaDevices.videoIn.selectedId,
+    mediaDevices.audioIn.selectedId,
+  ];
+  const video = usePreviewDevice(videoEnabled, videoId ?? "", "videoinput");
+  const audio = usePreviewDevice(audioEnabled, audioId ?? "", "audioinput");
+
+  const activeVideoId = video?.selectedDevice?.deviceId;
+  const activeAudioId = audio?.selectedDevice?.deviceId;
+  React.useEffect(() => {
+    const createChoices = (
+      enabled: boolean,
+      deviceId?: string
+    ): DeviceChoices => {
+      if (deviceId === undefined) {
+        return undefined;
+      }
+
+      return {
+        selectedId: deviceId,
+        enabled,
+      };
+    };
+
+    onUserChoicesChanged({
+      video: createChoices(videoEnabled, activeVideoId),
+      audio: createChoices(audioEnabled, activeAudioId),
```

Comment:
This has some latency to it right?
Choosing in the settings updates `mediaDevices` -> `mediaDevices.videoIn.selectedId` (and `mediaDevices.videoIn.selectedId` ) replace the track inside the usePreviewDevice hook. ->
Only when the new track is done initializing `activeVideoId` changes. -> the settings get updated.
not sure that is an issue.
But i think there is no other way to make the transfer of the permission request choice work...
Also it is unlikely some can start the call quick enough before the track is initialzied

Reply:
If I got you right, do you mean the case where someone selects a media and then clicks on the button before choices are getting initialized? - The `InCallView` is only created when the choices are not `undefined`. I.e. it's only on the render which has the user choices set the `InCallView` will be set.

I can imagine that perhaps there is a case where the user does not accept the dialog and instead presses the join button, in which case the old choices will be used. But this seems to be an expected behavior (i.e. if you have not selected anything, then the previous selection will be used).